### PR TITLE
Remove the compression command configuration, and only use zstandard

### LINKED ISSUE #430: Fully switch to zstd compression
We have decided to reduce complexity of the pipeline by supporting only one and the most efficient compression algorithm - [zstd](https://github.com/facebook/zstd). We should remove all the gzip/pigz related code from the core pipeline scripts and taskcluster kinds ( `COMPRESSION_CMD`, `COMPRESSION_EXT` env variables, --compression_cmd cli args etc.)

### PR DESCRIPTION
**Pull request by gregtatum**
Resolves #430

### CODE REVIEW

**Comment by gregtatum** [comment]
Ah, I missed a commit that fixes CI here.

File: pipeline/common/downloads.py
Comment (eu9ene) [inline pipeline/common/downloads.py:476]: Should we add unit tests for compress/decompress functions? They are quite important to work correctly now.

**Review by eu9ene — COMMENTED** [review]
Nice refactoring that removes a lot of unnecessary complexity from the code! I suggest adding unit tests for the new compression/decompression functions before merging.

File: pipeline/common/downloads.py
Comment (gregtatum) [reply pipeline/common/downloads.py:476]: Good idea. I'll do it. I think we're covered by integration tests, but it's always nice to be explicit on stuff like this.

**Review by eu9ene — APPROVED** [review]
lgtm, but I think @bhearsum should have a look too, since a lot of Taskcluster code is affected

**Review by bhearsum — APPROVED** [review]
This is great! Thanks for doing it. The taskcluster stuff looks very safe; the pipeline run here seems to confirm that there's no issues.

**Pull request merged**
